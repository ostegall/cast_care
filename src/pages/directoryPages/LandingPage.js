import { Text, View, Pressable } from 'react-native';
import styles from '../../styles';

export default function LandingPage({navigation}) { //Opening Page for user to choose between looking at general cast upkeep or diagnosin & solving cast issues
    return (
      <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
        <Pressable style={styles.button} 
              onPress = {() => {
                  navigation.navigate('Cast Info'); //navigates to page to look at general cast upkeep
              }}>
          <Text style={styles.text}>
              Click here to read general cast upkeep information
          </Text>
        </Pressable>
        <Pressable style={styles.button} 
              onPress = {() => {
                  navigation.navigate('Open Screen'); //navigates to page to diagnose and solve cast issues
              }}>
          <Text style={styles.text}>
              Click here to diagnose and solve cast issues
          </Text>
        </Pressable>      
      </View>
    )
  }