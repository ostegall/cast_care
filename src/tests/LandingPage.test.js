import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

import LandingPage from '../pages/directoryPages/LandingPage';

//Makes sure app component renders correctly without crashing
it("renders without crashing", () => {
  shallow(<LandingPage />);
});

//Snapshot test to ensure UI stays consistent
it('renders correctly', () => {
  const tree = renderer.create(<LandingPage />).toJSON();
  expect(tree).toMatchSnapshot();
});