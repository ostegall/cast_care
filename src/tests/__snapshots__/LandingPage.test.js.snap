// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 50,
          "fontWeight": "bold",
          "letterSpacing": 0.5,
          "textAlign": "center",
        }
      }
    >
      Welcome to Cast Care!
    </Text>
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 28,
            "letterSpacing": 0.5,
            "textAlign": "center",
          }
        }
      >
        What cast information would you like to find?
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "paddingBottom": 20,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4B9CD3",
            "borderRadius": 10,
            "elevation": 2,
            "flexDirection": "row",
            "height": 180,
            "justifyContent": "center",
            "minWidth": 88,
            "paddingLeft": 16,
            "paddingRight": 16,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 5,
            "width": 280,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 30,
              "letterSpacing": 0.25,
              "textAlign": "center",
            }
          }
        >
          General Cast Upkeep Information
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4B9CD3",
          "borderRadius": 10,
          "elevation": 2,
          "flexDirection": "row",
          "height": 180,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 280,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "letterSpacing": 0.25,
            "textAlign": "center",
          }
        }
      >
        Common Cast Issues
      </Text>
    </View>
  </View>
</View>
`;
